packages/storybook/src/nimble/select/select.mdx

```diff
@@ -52,6 +52,23 @@ the end of the prompt.
 
 <Canvas of={selectStories.placeholder} />
 
+### Dynamic Options
+
+There are scenarios that might require not all possible options to be available
+initially, such as when the set of all options is too large to be loaded at once. 
+In such cases, users are expected to provide filter text to narrow down the list 
+of options. The `Select` now offers the `filter-input` event to allow clients to
+listen for changes to the filter input. When accompanied with the `filterMode` set
+to `manual`, clients can perform their own filtering and provide the set of options
+to the `Select` dynamically.
+
+Typical steps for dynamic options are as follows:
+1. Set the `filterMode` to `manual` on the `Select`.
+1. Listen for the `filter-input` event on the `Select`.
+1. In the event handler, retrieve the filter text from the event and populate the DOM
+with the set of options that match the filter text by your own logic.
+    - It is important that one of the options in the DOM is the currently selected option.
+
```

Comment:
In the HLD we have a bunch of useful guidance about how clients should do filtering (copied below). Can you summarize the relevant points and add them here? I think setting `loading-visible` is particularly important.

----

IMPORTANT: When using the manual filter mode, clients are responsible for the following:

- Either adding/removing options to/from the DOM that do or do not match the filter and/or marking existing options as hidden when they don't match.
- The currently selected option is included in the set of options placed in the DOM regardless of match against filter (and be marked as hidden if it doesn't match the filter)
- If a placeholder option exists for the select, it should never be filtered out.

Notes:

- When the loading-visible attribute is set, we will display localizable text at the bottom of the dropdown (defaulting to "Loading"), along with the spinner icon.
- The Select will not peform any debouncing as the user types into the filter input. It is expected that clients can perform any debouncing that is needed easily at the app level.
- The filter-input event will be emitted for all filter modes (except none)

Groups

When using the filterMode: manual option along with [groups](https://github.com/ni/nimble/blob/main/packages/nimble-components/src/select/specs/option-groups-hld.md), clients will also need to determine the appropriate matching behavior against the ListOptionGroup elements. Our guidance will be that filter text should be able to be matched against any text in the group label, and when that matches, all options within that group should be visible. Conversely, the select will automatically hide groups with no items in them. This means clients only need to conditionally remove items, not groups (though it's fine if they manually remove empty groups too).